code/datums/components/spawner.dm

```diff
@@ -1,14 +1,19 @@
 /datum/component/spawner
+	/// List of mob types to spawn, picked randomly
 	var/mob_types = list(/mob/living/basic/carp)
-	var/spawn_time = 300 //30 seconds default
+	/// Time to wait between spawns
+	var/spawn_time = 30 SECONDS
+	/// List of weak references to mobs we have already created
 	var/list/spawned_mobs = list()
-	var/spawn_delay = 0
+	/// Time until we next spawn
+	COOLDOWN_DECLARE(spawn_delay)
```

Comment:
imo having COOLDOWNS be placed at the very end of the var list is the cleanest to my eyes if you wanna push it down there, no obligation though